google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +146,29 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    // ensures finite iteration
+    int maxDepth = 10;
+    while (throwable != null && maxDepth-- > 0) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
```

Comment:
Just in case, please also change this line to
(throwable.getMessage() != null && throwable.getMessage().contains("Connection closed prematurely")), as exceptions are not required to provide messages.